internal/util/kubernetes/config_watcher.go

```diff
@@ -34,44 +34,86 @@ func StartConfigMapWatcher(ctx context.Context, config *rest.Config) error {
 			watcher, err = client.CoreV1().ConfigMaps(namespace).Watch(ctx, metav1.ListOptions{
-				LabelSelector: configMapLabel,
+				LabelSelector: common.LabelKeyNumaplaneControllerConfig,
 			})
 			numaLogger.Error(err, "watcher channel closed, restarting watcher")
 			continue
 		}
+
 		configMap, ok := event.Object.(*corev1.ConfigMap)
 		if !ok {
 			numaLogger.Error(fmt.Errorf("failed to convert object to configmap"), "")
 		}
 
-		// Add or update the controller definition config based on a version if the configmap has the correct label
-		for _, v := range configMap.Data {
-			var controllerConfig config.NumaflowControllerDefinitionConfig
-			if err := yaml.Unmarshal([]byte(v), &controllerConfig); err != nil {
-				numaLogger.Error(err, "Failed to unmarshal controller config")
-				continue
-			}
-			// controller config definition is immutable, so no need to update the existing config
-			if event.Type == watch.Added {
-				config.GetConfigManagerInstance().GetControllerDefinitionsMgr().UpdateControllerDefinitionConfig(controllerConfig)
-			} else if event.Type == watch.Deleted {
-				config.GetConfigManagerInstance().GetControllerDefinitionsMgr().RemoveControllerDefinitionConfig(controllerConfig)
+		labelVal := configMap.Labels[common.LabelKeyNumaplaneControllerConfig]
+		switch labelVal {
+		case common.LabelValueNumaflowControllerDefinitions:
+			handleNumaflowControllerDefinitionsConfigMapEvent(ctx, configMap, event)
+		case common.LabelValueUSDEConfig:
+			if err := handleUSDEConfigMapEvent(configMap, event); err != nil {
+				numaLogger.Error(err, "error while handling event on USDE ConfigMap")
 			}
+		default:
+			numaLogger.Errorf(err, "the ConfigMap named '%s' is not supported", configMap.Name)
+		}
+	}
+}
+
+func handleNumaflowControllerDefinitionsConfigMapEvent(ctx context.Context, configMap *corev1.ConfigMap, event watch.Event) {
+	numaLogger := logger.FromContext(ctx)
+
+	// Add or update the controller definition config based on a version if the configmap has the correct label
+	for _, v := range configMap.Data {
+		var controllerConfig config.NumaflowControllerDefinitionConfig
+		if err := yaml.Unmarshal([]byte(v), &controllerConfig); err != nil {
+			numaLogger.Error(err, "failed to unmarshal Numaflow Controller Definitions config")
+			continue
+		}
+
+		// controller config definition is immutable, so no need to update the existing config
+		if event.Type == watch.Added {
+			config.GetConfigManagerInstance().GetControllerDefinitionsMgr().UpdateNumaflowControllerDefinitionConfig(controllerConfig)
+		} else if event.Type == watch.Deleted {
+			config.GetConfigManagerInstance().GetControllerDefinitionsMgr().RemoveNumaflowControllerDefinitionConfig(controllerConfig)
 		}
 	}
 }
+
+func handleUSDEConfigMapEvent(configMap *corev1.ConfigMap, event watch.Event) error {
+	usdeConfig := config.USDEConfig{}
+
+	err := yaml.Unmarshal([]byte(configMap.Data["pipelineSpecExcludedPaths"]), &usdeConfig.PipelineSpecExcludedPaths)
```

Comment:
does Delete case work okay? Does `configMap==nil`?

Reply:
When deleting the ConfigMap, the configMap argument on line 98 is still the previous ConfigMap. In any case, I added some nil checks (just in case) and I also moved the logic inside the event if-statement because I think it makes more sense in this case.

Reply:
perfect, thanks